File: frontend/projects/api/src/lib/assessment-item.ts
import type { AnswerDefinition } from './answer-definition';
import type { EvidenceMetadata } from './evidence-metadata';
import type { GradingDefinition } from './grading-definition';

export interface AssessmentItem {
  id: string;
  type: string;
  prompt: string;
  options: string[] | null;
  rationale: string | null;
  grading: GradingDefinition;
  answer: AnswerDefinition;
  evidence?: EvidenceMetadata;
}

File: backend/src/RepoFluent.Application/AssessmentItem.cs
namespace RepoFluent.Application;

public sealed record AssessmentItem(
    string Id,
    string Type,
    string Prompt,
    IReadOnlyList<string>? Options,
    string? Rationale,
    GradingDefinition Grading,
    AnswerDefinition Answer)
{
    public EvidenceMetadata? Evidence { get; init; }
}
